simplified-books-database/src/main/java/org/nypl/simplified/books/book_database/BookDatabaseEntry.kt

```diff
@@ -252,29 +252,36 @@ internal class BookDatabaseEntry internal constructor(
       contentTypes: Set<MIMEType>
     ) {
       for (contentType in contentTypes) {
-        for (formatDefinition in constructors.keys) {
-          val constructor = constructors[formatDefinition]!!
-          if (formatDefinition.supportedContentTypes().contains(contentType)) {
-            if (!existingFormats.containsKey(constructor.classType)) {
-              val bookID = owner.book.id
-              logger.debug("[{}]: instantiating format {} for content type {}",
-                bookID.brief(),
-                constructor.classType.simpleName,
-                contentType)
-
-              val params =
-                DatabaseFormatHandleParameters(
-                  context = context,
-                  bookID = owner.book.id,
-                  directory = ownerDirectory,
-                  onUpdated = onUpdate,
-                  entry = owner,
-                  contentType = contentType
-                )
-
-              existingFormats[constructor.classType] = constructor.constructor.invoke(params)
-              return
+        for ((format, constructor) in constructors) {
+          if (format.supports(contentType)) {
```

Comment:
The actual fix for SIMPLY-2943 is on this line: Instead of matching on an exact MIMEType using `.contains(type)` we match using the existing `.supports(type)` method (which is lenient and does a comparison on MIMEType.fullType only, excluding any extra parameters).